engine/src/main/java/org/terasology/rendering/nui/widgets/UIImageSlideshow.java

```diff
@@ -51,16 +52,30 @@
     @LayoutConfig
     private boolean infinite = true;
 
+    /**
+     * Whether the slideshow automatically switch images.
+     */
+    @LayoutConfig
+    private boolean auto = true;
+
     @Override
     public void onDraw(Canvas canvas) {
         if (currentImage.get() != null) {
```

Comment:
Can be replaced with `currentImage.isPresent()`

Reply:
isPresent is method from Optional. What do you mean? 
Do you propose to make *currentImage* optional?